Run backend on same process as frontend.

notoraptor opened the pull request:
@breuleux This is a pull request to remove call to sub-processes when compiling graph.

I will let all tests run to check if there are some problems.

notoraptor commented:
@abergeron @breuleux Updated. I think it's done, but I needed to temporarily deactivate some GPU tests because they fail and @ethancaballero seems to work on it on PR #349 :
- tests.frontends.test_all_ops.py::test_torch_scatter_broadcast_source
- tests/frontends/test_pytorch.py::test_module_2_layer_mlp_seq_fwd
- tests/frontends/test_pytorch.py::test_module_linear_seq_bwd

(fails seems related to pytorch 1.5)

Once we find a way to fix these tests, I could reactivate them here before merging.
NB: In his PR, @ethancaballero just removed those tests. Maybe can I do the same ?

abergeron commented on tests/test_parser.py:
```diff
@@ -220,37 +226,17 @@ def c5():
 
     try:
         parse(f)
-    except MyiaSyntaxError:
```
This doesn't have anything to do with the multi-process.  It is testing that exception printout for myia parser errors give the proper stack trace on output.

Can you revert that back to what it was (this whole file)?

notoraptor replied:
Hi @abergeron I can revert, but this will make tests fail.

notoraptor commented:
Hi @breuleux @abergeron ! Just to say that everything seems to work here.
- I fixed TVM using this branch : https://github.com/apache/incubator-tvm/compare/master...notoraptor:fix-excepthook
- then I compiled tvm packages and uploaded them on my own conda channel: https://anaconda.org/notoraptor/tvm and https://anaconda.org/notoraptor/tvm-libs 
- and I use those packages in this branch (see requirement files)

So, this PR still cannot be merged until some more official TVM packages are used, but apart from that, it seems fine.

I also removed uncovered code.

abergeron commented on myia/compile/backends/__init__.py:
```diff
@@ -52,11 +52,11 @@ def pytorch_default(device="cpu:0"):
 _backends = {
     "relay": (
-        channel_loader("myia.compile.backends.relay", "RelayBackendR"),
+        channel_loader("myia.compile.backends.relay", "RelayBackend"),
```
You might want to rename channel_loader to something else.

notoraptor replied:
Renamed to backend_loader.

abergeron commented on requirements-cpu.conda:
```diff
@@ -2,7 +2,7 @@ python>=3.7,<3.8a0
 numpy
 opt_einsum
-abergeron::tvm==0.7dev1+1.*
+notoraptor::tvm==0.7dev1+1.*
```
You need to change the version if you change the code (the +1 should be +2).  Your +1 should be the same as my +1.

notoraptor replied:
By the way, this is changes applied to tvm: https://github.com/apache/incubator-tvm/compare/master...notoraptor:fix-excepthook

Should I still recompile and rename my package to ` notoraptor::tvm==0.7dev1+2.*` ?

abergeron replied:
Yes, otherwise it can easily create problems where people don't know which version of the code they have exactly and it makes it harder to debug problems.

This is why it's not a good practice to release two packages with the same version and different code.

notoraptor replied:
Ok, so I will recompile and re-upload a tvm package with version `+2`!

notoraptor commented:
@abergeron rebased, channel_loader renamed to backend_loader, and yaml and ruamel removed from requirements.

abergeron commented:
I think you forgot an import with your rebase and it makes the random tests fail.

notoraptor commented:
Indeed, fixed!

abergeron commented:
I'm ok with the changes here, the only thing we should wait for is the TVM PR.

notoraptor commented:
@abergeron Ok !

notoraptor commented:
@abergeron updated!
